experimental/devices/inky/inky.go

```diff
@@ -0,0 +1,327 @@
+		{0x04, nil},                // Power on
+		{0x2c, []byte{0x3c}},       // VCOM Register, 0x3c = -1.5v?
+		{0x3c, []byte{0x00}},
+		{0x3c, []byte{byte(border)}}, // Border colour
+	}
+
+	for _, c := range init {
+		if err := d.sendCommand(c.cmd, c.data); err != nil {
+			return err
+		}
+	}
+
+	switch d.color {
+	case Black:
+		if err := d.sendCommand(0x32, blackLUT[:]); err != nil {
+			return err
+		}
+	case Red:
+		if err := d.sendCommand(0x32, redLUT[:]); err != nil {
+			return err
+		}
+	case Yellow:
+		if err := d.sendCommand(0x04, []byte{0x07}); err != nil { // Set voltage of VSH and VSL.
+			return err
+		}
+		if err := d.sendCommand(0x32, yellowLUT[:]); err != nil {
+			return err
+		}
+	}
+
+	h := make([]byte, 4)
+	binary.LittleEndian.PutUint16(h[2:], rows)
+	write := []struct {
+		cmd  byte
+		data []byte
+	}{
+		{0x44, []byte{0x00, cols/8 - 1}}, // Set RAM X Start/End
+		{0x45, h},                        // Set RAM Y Start/End
+		{0x43, []byte{0x00}},
+
+		{0x4e, []byte{0x00}},
+		{0x4f, []byte{0x00, 0x00}},
+		{0x24, black},
+
+		{0x43, []byte{0x00}},
+		{0x4f, []byte{0x00, 0x00}},
+		{0x26, red},
+
+		{0x22, []byte{0xc7}},
+	}
+
+	for _, c := range write {
+		if err := d.sendCommand(c.cmd, c.data); err != nil {
+			return err
+		}
+	}
+
+	d.busy.In(gpio.PullUp, gpio.FallingEdge)
+	defer d.busy.In(gpio.PullUp, gpio.NoEdge)
+	if err := d.sendCommand(0x20, nil); err != nil {
+		return err
+	}
+
+	d.busy.WaitForEdge(-1)
+
+	if err := d.sendCommand(0x10, []byte{0x01}); err != nil { // Enter deep sleep.
+		return err
+	}
+	return nil
+}
+
+func (d *Dev) reset() error {
+	d.r.Out(gpio.Low)
+	time.Sleep(100 * time.Millisecond)
+	d.r.Out(gpio.High)
+	time.Sleep(100 * time.Millisecond)
+
+	d.busy.In(gpio.PullUp, gpio.FallingEdge)
+	defer d.busy.In(gpio.PullUp, gpio.NoEdge)
+	if err := d.sendCommand(0x12, nil); err != nil { // Soft Reset
+		return fmt.Errorf("failed to reset inky: %v", err)
+	}
+	d.busy.WaitForEdge(-1)
+	return nil
+}
+
+func (d *Dev) sendCommand(command byte, data []byte) error {
+	d.dc.Out(gpio.Low)
+	if err := d.c.Tx([]byte{command}, nil); err != nil {
+		return fmt.Errorf("failed to send command %x to inky: %v", command, err)
+	}
+	if data != nil {
+		if err := d.sendData(data); err != nil {
+			return fmt.Errorf("failed to send data for command %x to inky: %v", command, err)
+		}
+	}
+	return nil
+}
+
+func (d *Dev) sendData(data []byte) error {
+	if len(data) > 4096 {
+		return fmt.Errorf("Sending more data than chunk size: %d > 4096", len(data))
+	}
+	d.dc.Out(gpio.High)
+	if err := d.c.Tx(data, nil); err != nil {
+		return fmt.Errorf("failed to send data to inky: %v", err)
+	}
+	return nil
+}
+
+func pack(bits []bool) ([]byte, error) {
+	if len(bits)%8 != 0 {
+		return nil, fmt.Errorf("len(bits) must be multiple of 8 but is %d", len(bits))
+	}
+
+	ret := make([]byte, len(bits)/8)
+	for i, b := range bits {
+		index := i / 8
+		shift := 7 - uint(i)%8
+		ret[index] |= (boolToByte(b) << shift)
```

Comment:
boolToByte is still not needed, since you don't need to set it when it's false, so the code can be simplified to:

```
if b {
  ret[i/8] |= 1 << (7 - uint(i)%8)
}
```

It's both faster and less complex to read.